plan/typeinfer_test.go

```diff
@@ -121,6 +121,8 @@ func (s *testPlanSuite) createTestCase4Columns() []typeInferTestCase {
 		{"c_int", mysql.TypeLong, charset.CharsetBin, mysql.BinaryFlag, 11, types.UnspecifiedLength},
 		{"c_char", mysql.TypeString, charset.CharsetUTF8, 0, 20, types.UnspecifiedLength},
 		{"c_enum", mysql.TypeEnum, charset.CharsetUTF8, 0, types.UnspecifiedLength, types.UnspecifiedLength},
+		// this doesn't pass
```

Comment:
refer to https://github.com/pingcap/tidb/issues/4165.
It has nothing related to this pr. I'll remove the comment.